### PR TITLE
[17.0][MIG] hr_timesheet_analytic_tag: Migration to 17.0

### PR DESCRIPTION
**Pull request by BernatObrador**
Module migrated to version 17.0

cc https://github.com/APSL 161924

@miquelalzanillas @lbarry-apsl @mpascuall @peluko00 @javierobcn @ppyczko please review

### CODE REVIEW

**Review by mpascuall — APPROVED** [review]
LGTM, tested in runboat

**Review by peluko00 — APPROVED** [review]
LGTM, code review done

**Review by ppyczko — APPROVED** [review]
LGTM, tested in runboat

File: hr_timesheet_analytic_tag/models/account_analytic_line.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    def _timesheet_preprocess(self, vals):
+        res = super()._timesheet_preprocess(vals)
+        for val in vals:
+            if val.get("task_id") and not val.get("tag_ids"):
+                task = self.env["project.task"].browse(val.get("task_id"))
+                val.update(tag_ids=[Command.set(task.analytic_tag_ids.ids)])
```
Comment (pedrobaeza) [inline hr_timesheet_analytic_tag/models/account_analytic_line.py]: Please use proper variable names that fit the signature upstream

```suggestion
    def _timesheet_preprocess(self, vals_list):
        res = super()._timesheet_preprocess(vals_list)
        for vals in vals_list:
            if vals.get("task_id") and not vals.get("tag_ids"):
                task = self.env["project.task"].browse(vals.get("task_id"))
                vals["tag_ids"] = [Command.set(task.analytic_tag_ids.ids)]
```

**Comment by BernatObrador** [comment]
@pedrobaeza Thanks for your contribution! The change has been made. Could you review it again, please?

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**